api/v1alpha1/adminnetworkpolicy_types.go

```diff
@@ -0,0 +1,290 @@
+/*
+Copyright 2022.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// AdminNetworkPolicy is the Schema for the adminnetworkpolicies API
+type AdminNetworkPolicy struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of AdminNetworkPolicy.
+	// +optional
+	Spec AdminNetworkPolicySpec `json:"spec,omitempty"`
+
+	// ANPStatus is the status to be reported by the implementation, this is not
+	// standardized in alpha and consumers should report what they see fit in
+	// relation to their AdminNetworkPolicy implementation
+	// +optional
+	Status AdminNetworkPolicyStatus `json:"status,omitempty"`
+}
+
+// AdminNetworkPolicyStatus defines the observed state of AdminNetworkPolicy
+type AdminNetworkPolicyStatus struct {
+	Conditions []metav1.Condition `json:"status"`
+}
+
+// AdminNetworkPolicySubject defines what objects the policy selects.
+// Exactly one of the `NamespaceSelector` or `NamespaceAndPodSelector` pointers
+// should be set.
+type AdminNetworkPolicySubject struct {
+	NamespaceSelector       *metav1.LabelSelector `json:"namespaceselector"`
```

Comment:
json tags should be camel case'd (e.g., namespaceAndPodSelector)

Reply:
Ack Fixed